From: Steve McIntyre <steve@einval.com>
Date: Sun, 18 Sep 2022 23:12:02 +0100
Subject: tests: Explicitly unset SOURCE_DATE_EPOCH before running fs tests

In some filesystem utils like mksquashfs, they will silently change
behaviour and cause timestamps to unexpectedly change. Build
environments like Debian's set SOURCE_DATE_EPOCH in the environment,
so remove it. Reproducible builds are good and useful for shipped
artifacts, but this causes build-time tests to fail.

Signed-off-by: Steve McIntyre <steve@einval.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 tests/util/grub-fs-tester.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/util/grub-fs-tester.in b/tests/util/grub-fs-tester.in
index bc14a05..69ef727 100644
--- a/tests/util/grub-fs-tester.in
+++ b/tests/util/grub-fs-tester.in
@@ -2,6 +2,9 @@
 
 set -e
 
+# We can't have this set, or filesystem tests will fail.
+unset SOURCE_DATE_EPOCH
+
 fs="$1"
 
 GRUBFSTEST="@builddir@/grub-fstest"
